### PR TITLE
add generic detection models

### PR DESCRIPTION
**Pull request by littletomatodonkey**
1. Combine open image v5 dataset and objects365 dataset to generate a new dataset, based on this we trained some networks: `cascade_rcnn_dcn_r101_vd_fpn_generic_server_side` and `cascade_rcnn_dcn_r50_vd_fpn_generic_server_side`
3. One can directly use the two models with 676 categories for inference as we provided the label map.

### CODE REVIEW

File: configs/rcnn_enhance/cascade_rcnn_dcn_r101_vd_fpn_2x_server_side_generic_pretrain.yml [hunk 1]
```diff
@@ -0,0 +1,218 @@
+architecture: CascadeRCNN
+max_iters: 360000
+snapshot_iter: 30000
+use_gpu: true
+log_smooth_window: 20
+log_iter: 20
+save_dir: output
+pretrain_weights: https://paddle-imagenet-models-name.bj.bcebos.com/bject_detection/cascade_rcnn_dcn_r101_vd_fpn_generic_server_side.tar
```
Comment (jerrywgz) [inline configs/rcnn_enhance/cascade_rcnn_dcn_r101_vd_fpn_2x_server_side_generic_pretrain.yml:8]: 这里的路径应该修改为分类的预训练？另外bject_detection->object_detection

File: configs/rcnn_enhance/cascade_rcnn_dcn_r101_vd_fpn_2x_server_side_generic_pretrain.yml [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+  - !PadBatch
+    pad_to_stride: 32
+    use_padded_im_info: false
+  batch_size: 1
+  shuffle: true
+  worker_num: 2
+  use_process: false
+
+
+TestReader:
```
Comment (jerrywgz) [inline configs/rcnn_enhance/cascade_rcnn_dcn_r101_vd_fpn_2x_server_side_generic_pretrain.yml:151]: testreader 统一放在在evalreader后面吧

File: ppdet/utils/generic_eval.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
+# Copyright (c) 2019 PaddlePaddle Authors. All Rights Reserved.
```
Comment (jerrywgz) [inline ppdet/utils/generic_eval.py:1]: 2019->2020

File: ppdet/utils/generic_det_label.map [hunk 4]
```diff
@@ -0,0 +1,676 @@
+38	Bowl	碗
```
Comment (jerrywgz) [inline ppdet/utils/generic_det_label.map:1]: 这里类别为中文吗？需要在py2 py3环境下都验证下

File: tools/infer.py [hunk 5]
```diff
@@ -126,10 +126,10 @@ def main():
         checkpoint.load_params(exe, infer_prog, cfg.weights)
 
     # parse infer fetches
-    assert cfg.metric in ['COCO', 'VOC', 'OID', 'WIDERFACE'], \
+    assert cfg.metric in ['COCO', 'VOC', 'OID', 'WIDERFACE', 'GENERIC'], \
```
Comment (jerrywgz) [inline tools/infer.py:129]: 这里需要在tools/export_model和deploy/python/infer中验证下能否正常导出模型并预测

**Comment by greathope** [comment]
@littletomatodonkey  where can I download these two weights?

**Comment by littletomatodonkey** [comment]
> @littletomatodonkey where can I download these two weights?

Thanks for your attention, we will open source the models after this PR is merged~

File: configs/rcnn_enhance/cascade_rcnn_dcn_r101_vd_fpn_2x_server_side_generic_pretrain.yml (hunk 1 above)
Comment (littletomatodonkey) [reply configs/rcnn_enhance/cascade_rcnn_dcn_r101_vd_fpn_2x_server_side_generic_pretrain.yml]: 好的， 这个yaml文件删除了~

File: configs/rcnn_enhance/cascade_rcnn_dcn_r101_vd_fpn_2x_server_side_generic_pretrain.yml (hunk 2 above)
Comment (littletomatodonkey) [reply configs/rcnn_enhance/cascade_rcnn_dcn_r101_vd_fpn_2x_server_side_generic_pretrain.yml]: done

File: ppdet/utils/generic_eval.py (hunk 3 above)
Comment (littletomatodonkey) [reply ppdet/utils/generic_eval.py]: done

File: ppdet/utils/generic_det_label.map (hunk 4 above)
Comment (littletomatodonkey) [reply ppdet/utils/generic_det_label.map]: 这里之后统一为了voc格式label了

File: tools/infer.py (hunk 5 above)
Comment (littletomatodonkey) [reply tools/infer.py]: 这里之后统一为了voc格式数据了，导出和预测于VOC相同

**Comment by littletomatodonkey** [comment]
> 针对677类这个模型，可以有一篇文档介绍下，然后在首页的README.md中链下。

在服务器端实用目标检测模型文档中补充了简介和模型链接~

File: README.md [hunk 6]
```diff
@@ -130,7 +130,7 @@
 - [车辆检测预训练模型](docs/featured_model/CONTRIB_cn.md)
 - [Objects365 2019 Challenge夺冠模型](docs/featured_model/champion_model/CACascadeRCNN.md)
 - [Open Images 2019-Object Detction比赛最佳单模型](docs/featured_model/champion_model/OIDV5_BASELINE_MODEL.md)
-- [服务器端实用目标检测模型](configs/rcnn_enhance/README.md): V100上速度20FPS时，COCO mAP高达47.8%。
+- [服务器端实用目标检测模型](configs/rcnn_enhance/README.md): V100上速度20FPS时，COCO mAP高达47.8%；此外也提供了包括676个类别的实用目标检测模型，适用于绝大部分使用场景。
```
Comment (qingqing01) [inline README.md:133]: 单独加一栏：

大规模检测预训练模型 ？

Comment (littletomatodonkey) [reply README.md]: 已添加~

Comment (littletomatodonkey) [reply README.md]: 已添加~

**Comment by littletomatodonkey** [comment]
> @littletomatodonkey where can I download these two weights?

Hi, you can refer to this doc for more pretrained weights: https://github.com/PaddlePaddle/PaddleDetection/blob/master/docs/featured_model/LARGE_SCALE_DET_MODEL_en.md

File: tools/infer.py (hunk 5 above)
Comment (lazyn1997) [reply tools/infer.py]: 我用export_model.py导出模型：
python .\tools\export_model.py -c .\configs\rcnn_enhance\generic\casc
ade_rcnn_dcn_r50_vd_fpn_gen_server_side.yml -o weights=.\configs\rcnn_enhance\generic\cascade_rcnn_dcn_r50_vd_fpn_g
en_server_side.pdparams
或者infer.py预测：
python tools/infer.py -c configs/rcnn_enhance/generic/cascade_rcnn_dc
n_r50_vd_fpn_gen_server_side.yml --infer_img=E:/Documents/postgraduate/fish_recognition_yolov4/fish/first/JPEGImage
s/237_222.png -o weights=configs/rcnn_enhance/generic/cascade_rcnn_dcn_r50_vd_fpn_gen_server_side.pdparams
时，使用https://github.com/PaddlePaddle/PaddleDetection/blob/release/2.0/static/docs/featured_model/LARGE_SCALE_DET_MODEL.md链接的配置文件会报错如下：
yaml.constructor.ConstructorError: could not determine a constructor for the tag '!DecodeImage'
  in ".\configs\rcnn_enhance\generic\cascade_rcnn_dcn_r50_vd_fpn_gen_server_side.yml", line 122, column 5
能否告知是什么原因？

**Pull request merged**